app/renderer/js/pages/preference/new-server-form.ts

```diff
@@ -58,16 +58,16 @@ class NewServerForm extends BaseComponent {
 		this.$newServerUrl = this.$newServerForm.querySelectorAll('input.setting-input-value')[0] as HTMLInputElement;
 	}
 
-	submitFormHandler(): void {
+	async submitFormHandler(): Promise<void> {
 		this.$saveServerButton.innerHTML = 'Connecting...';
-		DomainUtil.checkDomain(this.$newServerUrl.value).then(serverConf => {
-			DomainUtil.addDomain(serverConf).then(() => {
-				this.props.onChange(this.props.index);
-			});
-		}, errorMessage => {
+		try {
+			const serverConf = await DomainUtil.checkDomain(this.$newServerUrl.value);
+			await DomainUtil.addDomain(serverConf);
+			this.props.onChange(this.props.index);
+		}	catch (errorMessage) {
```

Comment:
This previously caught errors from `checkDomain` only, which probably makes more sense.